simple_certmanager/forms.py

```diff
@@ -12,6 +12,11 @@ class CertificateAdminForm(forms.ModelForm):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+        # When a certificate/key pair is first created, the instance is None
+        if kwargs.get("instance") is None:
```

Comment:
after `super().__init__` you should be able to use `self.instance`, if no explicit instance was provided, it will be an empty model instances (and then I think it'll fall in the `except` flow. If it doesn't, you can check using `self.instance.pk` for an unsaved object.

Reply:
It does indeed fall into the except flow, but with `ValueError: The 'public_certificate' attribute has no file associated with it.`. So I added the extra error to catch